### PR TITLE
DOC: stats.circstd: add reference, notes, comments

### LINKED ISSUE #10096: Add literature reference for circstd (and circvar?)
I'm trying to figure out what reference to cite, should I use Scipy's `circstd()` function. Unfortunately, no reference is currently given, which renders it unusable for me to use in research.

As far as I could find, the closest formulation of the [current implementation  of `circstd`](https://github.com/scipy/scipy/blob/c42462ab9c697d3e14a7cc232b967c50886438a9/scipy/stats/morestats.py#L3281) is Mardia (1972). Although it's not a literal translation into Python, I think it is equivalent, and the only source I found that *scales* the result according to the range of input values. (I can unpack this in detail, if somebody wants me to; note that e.g. Zar (2010) omits the scaling, although referring to Mardia, weirdly enough).

While I'm trying to focus on the circular standard deviation here, note that if `circstd()` was indeed based on Mardia, a secondary issue might arise regarding `circvar()`: The [current implementation of `circvar`](https://github.com/scipy/scipy/blob/c42462ab9c697d3e14a7cc232b967c50886438a9/scipy/stats/morestats.py#L3239) is a scaling of `2 * log(1/R)`, which is *inconsistent* with both Mardia and Zar (who define it as `1 - R`), and further authors (and thus software packages), as pointed out in https://github.com/scipy/scipy/pull/5747.

@teoliphant, the current implementation of both functions goes back to your PR https://github.com/scipy/scipy/pull/140, maybe you remember the reference you based them on?

References  
Mardia, K. V. (1972). *Statistics of directional data (Z. W. Birnbaum & E. Lukacs, Eds.)*. London: Academic Press.  
Zar, J. H. (2010). *Biostatistical analysis (5. ed).* Upper Saddle River, NJ: Prentice Hall.

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
closes gh-10096

#### What does this implement/fix?
This adds a literature reference for `stats.circstd`. 
It also adds comments indicating the equations from the reference that correspond with each line of code, includes the definition of the statistic in the notes, and demonstrates that the property originally mentioned in the notes (similar to linear standard deviation in the small angle limit) in the example.

image

#### Additional information
A separate PR addressing gh-5747 will add the reference for `stats.circvar`.

### CODE REVIEW

File: scipy/stats/_morestats.py [hunk 1]
```diff
@@ -3625,28 +3625,47 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
+        R = np.sqrt(C**2 + S**2)
+        l = 2*np.pi / (high-low)
+        circstd = np.sqrt(-2*np.log(R)) / l
+
+    In the limit of small angles, it returns a number close to the 'linear'
+    standard deviation.
+
+    References
+    -----
+    .. [1] Mardia, K. V. (1972). 2. *In Statistics of Directional Data*
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3644]:
```suggestion
    .. [1] Mardia, K. V. (1972). 2. In *Statistics of Directional Data*
```

Comment (mdhaber) [inline scipy/stats/_morestats.py:3644]:
```suggestion
    References
    ----------
    .. [1] Mardia, K. V. (1972). 2. In *Statistics of Directional Data*
```

File: scipy/stats/_morestats.py [hunk 2]
```diff
@@ -3625,28 +3625,47 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
+        l = 2*np.pi / (high-low)
+        circstd = np.sqrt(-2*np.log(R)) / l
+
+    In the limit of small angles, it returns a number close to the 'linear'
+    standard deviation.
+
+    References
+    ----------
+    .. [1] Mardia, K. V. (1972). 2. In *Statistics of Directional Data*
+       (pp. 18–24). Academic Press.
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3645]:
```suggestion
       (pp. 18-24). Academic Press.
```

**Comment by mdhaber** [comment]
@JoKeys how does this look to you?

File: scipy/stats/_morestats.py [hunk 3]
```diff
@@ -3646,28 +3646,47 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
+        l = 2*np.pi / (high-low)
+        circstd = np.sqrt(-2*np.log(R)) / l
+
+    In the limit of small angles, it returns a number close to the 'linear'
+    standard deviation.
+
+    References
+    ----------
+    .. [1] Mardia, K. V. (1972). 2. In *Statistics of Directional Data*
+       (pp. 18-24). Academic Press.
```
Comment (tupui) [inline scipy/stats/_morestats.py:3666]: I would add the DOI. (We should think about this more often. We also have `:arxiv:`)
```suggestion
    .. [1] Mardia, K. V. (1972). 2. In *Statistics of Directional Data*
       (pp. 18-24). Academic Press. :doi:`10.1016/C2013-0-07425-7`.
```

**Review by tupui — COMMENTED** [review]
LGTM, just a suggestion to add the DOI.

(In general we are missing a tutorial on these circ and geom stats.)

**Comment by mdhaber** [comment]
> (In general we are missing a tutorial on these circ and geom stats.)

I think that they should not be advertised with a tutorial in their current state. For instance, the calculation of `stats.circvar` is incorrect according to every reference we've found, which I'll fix when this merges.

**Comment by tupui** [comment]
> > (In general we are missing a tutorial on these circ and geom stats.)
> 
> I think that they should not be advertised with a tutorial in their current state. For instance, the calculation of `stats.circvar` is incorrect according to every reference we've found, which I'll fix when this merges. There is also very little here. We could also consider deprecating it all and just relying on [`astropy`](https://docs.astropy.org/en/stable/stats/circ.html) for this stuff. I'm conflicted.

Mmmm yes I also saw that `astropy` had this. But to me it seems like they did it because we lacked such methods. We can raise this on an issue/mailing list.

**Comment by tupui** [comment]
Doc renders good, merging. Thanks again Matt.

**Comment by JoKeyser** [comment]
For the record: @mdhaber thank you, from me as well, looks great.

**Comment by mdhaber** [comment]
Great, thanks @JoKeyser @tupui!

**Pull request merged**